### DISCUSSION TITLE
模板中使用citet引用英文文献，文中显示“et al.”，如何调整模板使之显示“等”

### DISCUSSION THREAD

**Discussion by Sapphir3** [tuna/thuthesis#874] [Q&A] [answered]
按照国标，当引用中文文献时显示“XX 等”，应用英文文献时显示“XX et al.”,现在的模板也是这么显示的。但是按照老师要求，即使是引用英文文献，使用citet命令也要在正文中显示“XX 等”。请问怎么调整latex模板实现这个功能？

**Comment by zepinglee** [comment]
你用的是哪种编译方法？`bibtex` (+ `natbib`) 还是 `biblatex`？

研究生院的《指南》第 20 页也是这么要求的，你应该先 argue 一下。

<img width="749" alt="Screenshot 2023-05-10 at 18 21 13" src="https://github.com/tuna/thuthesis/assets/12290822/a24b3db1-a126-4d41-9b5f-7a95f18795a4">

**Reply by Sapphir3** [reply]
首先感谢回复~
我用的是bibtex编译的。
有说法是17年之后的国标有改动，或者是高校在国标的基础上对格式要求有微调。据我了解，确实有其他高校已经在模板上做了修改，使用“XX 等”的格式。
我个人的情况是，有两位评审老师都给出了这个意见，所以我就想问一下对于thuthesis支不支持这两种格式变化的选项，如果目前还没有的话，怎么样能比较方便的修改格式？

**Reply by zepinglee** [reply]
还有一个问题，你是用的 numeric 还是 author-date 样式？

**Reply by note286** [reply]
有的评审专家真的是，提的一些格式上的问题完全与学校要求相悖，让学生很难做。

我这边有同学收到的意见是目录不能有三级标题，而学校要求目录有三级标题，就离谱。

**Reply by Sapphir3** [reply]
我用的是numeric样式。

**Reply by zepinglee** [reply, marked as answer]
> 我用的是numeric样式。

那也能理解。《指南》和国标的规定是针对 author-date 样式，其实没有规定 numeric 样式对应的 `\citet` 使用“et al.”还是“等”。

你可以将这里 <https://github.com/tuna/thuthesis/blob/e26380bf6979924acfc74819a2faf5b3ab721d48/thuthesis-numeric.bst#L194-L199> 改成

```
FUNCTION {citation.et.al} { "等" }
```

**Reply by Sapphir3** [reply]
好滴，非常感谢，我已经修改成功了~

**Comment by sherlockyzd** [comment]
如果我要使用citet时：张昆 等 (1994); Merkt 等 (1995) ，但是citep时使用 (张昆 等, 1994;
Mellinger et al., 1996)
Bibtex编译器好像不太行…………

**Reply by zepinglee** [reply]
1. 著者-出版年制建议使用 biblatex，具体参考 [biblatex-gb7714-2015](http://mirrors.ctan.org/macros/latex/contrib/biblatex-contrib/biblatex-gb7714-2015/biblatex-gb7714-2015.pdf) 的文档。

2. 研究生《手册》要求使用“et al.”。
<img width="750" alt="Screenshot 2023-10-18 at 11 43 51" src="https://github.com/tuna/thuthesis/assets/12290822/2f15fe7d-8fb4-4830-936e-1793d1a97e53">

**Reply by sherlockyzd** [reply]
感谢！但是说实话这个排版不优美……我尝试了该改bst文件让他判别我使用的是citet还是citep好像不太行……还是放弃了

**Reply by zepinglee** [reply]
这个其实是 GB/T 7714—2015 的要求。另外 bibtex 不能对中文姓名自动排序，所以建议使用 biblatex 方案。
